auth/store.go

```diff
@@ -408,9 +396,24 @@ func (as *authStore) UserAdd(r *pb.AuthUserAddRequest) (*pb.AuthUserAddResponse,
 		}
 	}
 
+	var decodedPassword []byte
+	var err error
+
+	if !options.NoPassword {
+		if r.Password != "" && r.HashedPassword == "" {
```

Comment:
thanks. UserChangePassword also has the following repeated code(14 lines), now could we wrap it  into a function?

Reply:
Got it, let me update here later.

Reply:
fixed in the latest version, could you take a look?

Reply:
lgtm. thanks.